src/modules/adaptors/Frax/CollateralFTokenAdaptorV2.sol

```diff
@@ -0,0 +1,261 @@
+     * @param adaptorData the collateral asset deposited into Fraxlend
+     */
+    function balanceOf(bytes memory adaptorData) public view override returns (uint256) {
+        IFToken fraxlendPair = abi.decode(adaptorData, (IFToken));
+        return _userCollateralBalance(fraxlendPair, msg.sender);
+    }
+
+    /**
+     * @notice Returns collateral asset
+     */
+    function assetOf(bytes memory _adaptorData) public view override returns (ERC20) {
+        IFToken fraxlendPair = abi.decode(_adaptorData, (IFToken));
+        return _userCollateralContract(fraxlendPair);
+    }
+
+    /**
+     * @notice This adaptor returns collateral, and not debt.
+     */
+    function isDebt() public pure override returns (bool) {
+        return false;
+    }
+
+    //============================================ Strategist Functions ===========================================
+
+    /**
+     * @notice Allows strategists to add collateral to the respective cellar position on FraxLend, enabling borrowing.
+     * @param _fraxlendPair The specified Fraxlend Pair
+     * @param _collateralToDeposit The amount of collateral to add to Fraxlend Pair position
+     */
+    function addCollateral(IFToken _fraxlendPair, uint256 _collateralToDeposit) public {
+        ERC20 _collateralToken = _userCollateralContract(_fraxlendPair);
+        _validateFToken(_fraxlendPair);
+
+        uint256 amountToDeposit = _maxAvailable(_collateralToken, _collateralToDeposit);
+        address fraxlendPair = address(_fraxlendPair);
+        _collateralToken.safeApprove(fraxlendPair, amountToDeposit);
+        _addCollateral(_fraxlendPair, amountToDeposit);
+
+        // Zero out approvals if necessary.
+        _revokeExternalApproval(_collateralToken, fraxlendPair);
+    }
+
+    /**
+     * @notice Allows strategists to remove collateral from the respective cellar position on FraxLend.
+     * @param _collateralAmount The amount of collateral to remove from fraxlend pair position
+     * @param _fraxlendPair The specified Fraxlend Pair
+     */
+    function removeCollateral(uint256 _collateralAmount, IFToken _fraxlendPair) public {
+        // remove collateral
+        _removeCollateral(_collateralAmount, _fraxlendPair);
+        uint256 _exchangeRate = _updateExchangeRate(_fraxlendPair); // need to calculate LTV
+        // Check if borrower is insolvent (AKA they have bad LTV), revert if they are
+        if (minimumHealthFactor > (_isSolvent(_fraxlendPair, _exchangeRate))) {
+            revert CollateralFTokenAdaptor__HealthFactorTooLow(address(_fraxlendPair));
+        }
+    }
+
+    /**
+     * @notice Caller calls `updateExchangeRate()` on specified FraxlendV2 Pair
+     * @param fraxlendPair The specified FraxLendPair
+     * @return exchangeRate needed to calculate the current health factor
+     */
+    function _getExchangeRate(IFToken fraxlendPair) internal virtual returns (uint256 exchangeRate) {
```

Comment:
Move this down with the other helper functions

Reply:
oh wait actually this is old, looks like you can just delete this function

Reply:
Done